spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientConfigUtils.java

```diff
@@ -154,8 +155,8 @@ static MultipleSourcesContainer configMapsDataByLabels(CoreV1Api client, String
 	 *     4. gather all the names of the secrets + decoded data they hold
 	 * </pre>
 	 */
-	static MultipleSourcesContainer secretsDataByName(CoreV1Api client, String namespace, Set<String> sourceNames,
```

Comment:
this is a change I've done in multiple places. Basically, imho, if you specify:

```
spring:
   cloud:
      kubernetes:
         secrets:
           sources:
               - name: a
                 strict-for-profiles:
                    - dev
                    - prod

```

and there are 3 active profiles at the moment: `k8s, dev, prod`, then we read secret `a-k8s` first, then `a-dev`, then `a-prod` - in that exact order.